framework/src/modules/nft/method.ts

```diff
@@ -203,4 +204,137 @@ export class NFTMethod extends BaseMethod {
 
 		return false;
 	}
+
+	public async lock(methodContext: MethodContext, module: string, nftID: Buffer): Promise<void> {
+		const nftStore = this.stores.get(NFTStore);
+
+		const nftExists = await nftStore.has(methodContext, nftID);
+
+		if (!nftExists) {
+			this.events.get(LockEvent).error(
+				methodContext,
+				{
+					module,
+					nftID,
+				},
+				NftEventResult.RESULT_NFT_DOES_NOT_EXIST,
+			);
+
+			throw new Error('NFT substore entry does not exist');
+		}
+
+		const owner = await this.getNFTOwner(methodContext, nftID);
+
+		if (owner.length === LENGTH_CHAIN_ID) {
+			this.events.get(LockEvent).error(
+				methodContext,
+				{
+					module,
+					nftID,
+				},
+				NftEventResult.RESULT_NFT_ESCROWED,
+			);
+
+			throw new Error('NFT is escrowed to another chain');
+		}
+
+		const userStore = this.stores.get(UserStore);
+		const userKey = userStore.getKey(owner, nftID);
+		const userData = await userStore.get(methodContext, userKey);
+
+		if (userData.lockingModule !== NFT_NOT_LOCKED) {
+			this.events.get(LockEvent).error(
+				methodContext,
+				{
+					module,
+					nftID,
+				},
+				NftEventResult.RESULT_NFT_LOCKED,
+			);
+
+			throw new Error('NFT is already locked');
+		}
+
+		userData.lockingModule = module;
+
+		await userStore.set(methodContext, userKey, userData);
+
+		this.events.get(LockEvent).log(methodContext, {
+			module,
+			nftID,
+		});
+	}
+
+	public async unlock(methodContext: MethodContext, module: string, nftID: Buffer): Promise<void> {
+		const nftStore = this.stores.get(NFTStore);
+
+		const nftExists = await nftStore.has(methodContext, nftID);
+
+		if (!nftExists) {
+			this.events.get(LockEvent).error(
+				methodContext,
+				{
+					module,
+					nftID,
+				},
+				NftEventResult.RESULT_NFT_DOES_NOT_EXIST,
+			);
+
+			throw new Error('NFT substore entry does not exist');
+		}
+
+		const owner = await this.getNFTOwner(methodContext, nftID);
+
+		if (owner.length === LENGTH_CHAIN_ID) {
+			this.events.get(LockEvent).error(
+				methodContext,
+				{
+					module,
+					nftID,
+				},
+				NftEventResult.RESULT_NFT_ESCROWED,
+			);
+
+			throw new Error('NFT is escrowed to another chain');
+		}
```

Comment:
This check implies that NFT can't be unlocked because it is escrowed to another chain. But, if it is escrowed to another chain, then it wasn't locked in the first place (locked NFT can't be escrowed).
I would therefore delete this condition because it implies that NFT can't be unlocked because it is escrowed, while in reality it can't be unlocked because it is not locked, so it can be a bit misleading to have this kind of error here.
BUT: If you think that providing escrowing information here is important, you can leave it but then I would throw an error `NFT is escrowed to another chain and it is not locked` or something like that.

Reply:
Hmm, 🤔 added the check to indirectly achieve the same effect as calling `getLockingModule` as LIP suggests, as `getLockingModule` throw error if NFT is escrowed, perhaps not relevant for this case as the lock operation won't happen on an escrowed NFT, so removing it makes sense @Incede  ?

Reply:
@mjerkov Do you mean the `unlock` function can never be called when a NFT is escrowed ? If yes then this check can be removed otherwise I think it should happen nonetheless. Personally I'd have simply called the `getLockingModule` function here similar to LIP specification, but maybe the error message `NFT is escrowed to another chain` coming from `getLockingModule` wouldn't fit here ?

Reply:
Yes, if NFT is escrowed then it's not locked so there's no sense in calling the `unlock` function, but it's not a problem if this stays. Feel free to resolve...

Reply:
Kept the logic to throw an error if NFT is escrowed to have the same effect as calling getLockingModule, however removed the event log in this [commit](https://github.com/LiskHQ/lisk-sdk/pull/8561/commits/de66f9eccfb95788b0fed6dc602da425988aa2a2).